src/goSuggest.ts

```diff
@@ -199,7 +199,15 @@ export class GoCompletionItemProvider implements vscode.CompletionItemProvider {
 										paramSnippets.push('${' + (i + 1) + ':' + param + '}');
 									}
 								}
-								item.insertText = new vscode.SnippetString(suggest.name + '(' + paramSnippets.join(', ') + ')');
+								let newSnippetString = null;
+								// Avoid adding snippet for function suggest when cursor is followed by ()
+								// i.e: met() -> method()()
+								if (lineText.substr(position.character, 2) === '()') {
+									newSnippetString = new vscode.SnippetString(suggest.name);
```

Comment:
@lggomez 

We don't need to create a `vscode.SnippetString` in this case.
When `item.insertText` is not set, then `item.label` is used by the auto-completion feature.
See https://github.com/Microsoft/vscode/blob/1.23.0/src/vs/vscode.d.ts#L3047-L3052